src/event.js

```diff
@@ -504,6 +565,77 @@ jQuery.event = {
 	}
 };
 
+// Ensure the presence of an event listener that handles manually-triggered
+// synthetic events by interrupting progress until reinvoked in response to
+// *native* events that it fires directly, ensuring that state changes have
+// already occurred before other listeners are invoked.
+function leverageNative( el, type, forceAdd, allowAsync ) {
+
+	// Setup must go through jQuery.event.add
+	if ( forceAdd ) {
+		jQuery.event.add( el, type, forceAdd );
+		return;
+	}
+
+	// Register the controller as a special namespace-universal handler
+	dataPriv.set( el, type, forceAdd );
+	jQuery.event.add( el, type, {
+		namespace: false,
+		handler: function( event ) {
+			var maybeAsync, result,
+				saved = dataPriv.get( this, type );
+
+			// Interrupt processing of the outermost synthetic .trigger()ed event
```

Comment:
It's essentially a summary of the relevant part of `leverageNative` strategy... when a (outer) synthetic event is triggered, we want to interrupt propagation (because focus/checkedness states are wrong) and replace it with a possibly-asynchronous native event (which will update them before invoking listeners). The handling of that native event then involves replacing its further propagation with an _inner_ synthetic event (for forwarding data from the outer trigger) that runs synchronously and is **not** interrupted, essentially picking up where outermost handling leaves off.

Reply:
OK, I guess it'd be hard to explain it better here.